cmd/shellNode.go

```diff
@@ -0,0 +1,210 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2instanceconnect"
+
+	"github.com/instructure-bridge/truss-cli/truss"
+)
+
+var shellNodeCmd = &cobra.Command{
+	Use:   "node [node-name]",
+	Short: "Launch a shell on a Truss node.",
+	Long: `Launch a shell on a Truss node via SSH.
+
+	Usage:
+
+	truss -e [env] shell node [instance-id]
+	truss -e [env] shell node [node-name]
+
+	Examples:
+
+  truss -e nonprod-cmh shell node i-0b35c43cd48ab85ee
+	truss -e nonprod-cmh shell node ip-10-12-11-61.us-east-2.compute.internal`,
+	Args: cobra.MaximumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		region, err := getAWSRegionFromKubeconfig()
+
+		if err != nil {
+			log.Fatal("Error retrieving AWS region from Kubeconfig:", err.Error())
```

Comment:
I think the convention with these `RunE` commands is to return error rather than ejecting here. I think Cobra ultimately does the right thing with it (displays the error message and command usage).

If we want to add more context to the errors, this https://github.com/pkg/errors might be the right move

Reply:
bubbling all these up with context added when it seemed appropriate with `errors.Wrap`